packages/core/src/interpreter.ts

```diff
@@ -1,16 +1,11 @@
 import { Condition } from './Condition';
 
-type ArgsExceptLast<F extends (...args: any[]) => any> =
-  F extends (a: any, c: any) => any
-    ? Parameters<(condition: Condition) => 0>
-    : F extends (a: any, b: any, c: any) => any
-      ? Parameters<(condition: Condition, value: Parameters<F>[1]) => 0>
-      : Parameters<(
-        condition: Condition,
-        value: Parameters<F>[1],
-        options: Parameters<F>[2],
-        ...args: unknown[]
-      ) => 0>;
```

Comment:
in the past, ts didn't support labels for tuples and in order to transfer proper argument names into inferred functions, I needed to use this construct `Parameters<(condition: Condition, value: Parameters<F>[1]) => 0>` because then this tuple preserverd the names in the function. without this it was just `(arg_0: condition, arg_1: ....)`

Reply:
`ArgsExceptLast` is now a generic type for an arbitairy function and just strips the last Argument.
This should still work, because `createInterpreter` enforces that the passed function must extend `AnyInterpreter`, which is required to have `Condition` as the first argument.

Reply:
CompoundCondition extends Condition, so technically it satisfies the restriction but then it doesn’t allow to pass Condition to interpret nested conditions. It forces to pass CompoundCondition